plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/FlushMetadataCacheProcedure.java

```diff
@@ -0,0 +1,112 @@
+import static java.lang.invoke.MethodHandles.lookup;
+import static java.util.Objects.requireNonNull;
+
+public class FlushMetadataCacheProcedure
+        implements Provider<Procedure>
+{
+    private static final String PROCEDURE_NAME = "flush_metadata_cache";
+
+    private static final String PARAM_SCHEMA_NAME = "SCHEMA_NAME";
+    private static final String PARAM_TABLE_NAME = "TABLE_NAME";
+
+    private final HiveMetastoreFactory metastoreFactory;
+    private final Optional<CachingHiveMetastore> cachingHiveMetastore;
+    private final TransactionLogAccess transactionLogAccess;
+
+    private static final MethodHandle FLUSH_HIVE_METASTORE_CACHE;
+
+    static {
+        try {
+            FLUSH_HIVE_METASTORE_CACHE = lookup().unreflect(FlushMetadataCacheProcedure.class.getMethod("flushMetadataCache", ConnectorSession.class, String.class, String.class));
+        }
+        catch (ReflectiveOperationException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    @Inject
+    public FlushMetadataCacheProcedure(
+            HiveMetastoreFactory metastoreFactory,
+            Optional<CachingHiveMetastore> cachingHiveMetastore,
+            TransactionLogAccess transactionLogAccess)
+    {
+        this.metastoreFactory = requireNonNull(metastoreFactory, "metastoreFactory is null");
+        this.cachingHiveMetastore = requireNonNull(cachingHiveMetastore, "cachingHiveMetastore is null");
+        this.transactionLogAccess = requireNonNull(transactionLogAccess, "transactionLogAccess is null");
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                "system",
+                PROCEDURE_NAME,
+                ImmutableList.of(
+                        new Procedure.Argument(PARAM_SCHEMA_NAME, VARCHAR, false, null),
+                        new Procedure.Argument(PARAM_TABLE_NAME, VARCHAR, false, null)),
+                FLUSH_HIVE_METASTORE_CACHE.bindTo(this),
+                true);
+    }
+
+    public void flushMetadataCache(ConnectorSession session, String schemaName, String tableName)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            doFlushMetadataCache(session, Optional.ofNullable(schemaName), Optional.ofNullable(tableName));
+        }
+    }
+
+    private void doFlushMetadataCache(ConnectorSession session, Optional<String> schemaName, Optional<String> tableName)
+    {
+        if (schemaName.isEmpty() && tableName.isEmpty()) {
+            cachingHiveMetastore.ifPresent(CachingHiveMetastore::flushCache);
+            transactionLogAccess.flushCache();
+        }
+        else if (schemaName.isPresent() && tableName.isPresent()) {
+            HiveMetastore metastore = metastoreFactory.createMetastore(Optional.of(session.getIdentity()));
+            metastore.getTable(schemaName.get(), tableName.get())
+                    .ifPresent(table -> {
```

Comment:
For a table which doesn't exist, the procedure will still silently succeed. I see the same behaviour for hive's `flush_metadata_cache`. I think this is ok, but want to point it out in case it was not intentional.

Reply:
I followed the existing procedure's behavior, but it might be counterintuitive because we need to get the table location in case of Delta. I changed to throw an exception instead.